src/sentry/services/hybrid_cloud/rpc.py

```diff
@@ -504,60 +504,136 @@ def result_to_dict(value: Any) -> Any:
+            else:
+                response = self._fire_request(headers, data)
+            metrics.incr(
+                "hybrid_cloud.dispatch_rpc.response_code", tags={"status": response.status_code}
+            )
 
-def _fire_request(url: str, path: str, body: Any) -> requests.Response:
-    # TODO: Performance considerations (persistent connections, pooling, etc.)?
-    data = json.dumps(body).encode(_RPC_CONTENT_CHARSET)
+            if response.status_code == 200:
+                return response.json()
+            self._raise_from_response_status_error(response)
 
-    signature = generate_request_signature(path, data)
-    headers = {
-        "Content-Type": f"application/json; charset={_RPC_CONTENT_CHARSET}",
-        "Authorization": f"Rpcsignature {signature}",
-    }
-    return requests.post(url, headers=headers, data=data)
+    @contextmanager
+    def _open_request_context(self):
+        timer = metrics.timer(
+            "hybrid_cloud.dispatch_rpc.duration",
+            tags={"service": self.service_name, "method": self.method_name},
+        )
+        span = sentry_sdk.start_span(
+            op="hybrid_cloud.dispatch_rpc",
+            description=f"rpc to {self.service_name}.{self.method_name}",
+        )
+        with span, timer:
+            yield
+
+    def _raise_from_response_status_error(self, response: requests.Response) -> NoReturn:
+        if in_test_environment():
+            if response.status_code == 500:
+                raise RpcRemoteException(
+                    f"Error invoking rpc at {self.path!r}: check error logs for more details"
+                )
+            detail = response.json()["detail"]
+            raise RpcRemoteException(
+                f"Error ({response.status_code} status) invoking rpc at {self.path!r}: {detail}"
+            )
+        # Careful not to reveal too much information in production
+        if response.status_code == 403:
+            raise RpcRemoteException("Unauthorized service access")
+        if response.status_code == 400:
+            raise RpcRemoteException("Invalid service request")
+        raise RpcRemoteException(f"Service unavailable ({response.status_code} status)")
+
+    def _fire_test_request(self, headers: Mapping[str, str], data: bytes) -> Any:
+        from django.test import Client
+
+        from sentry.db.postgres.transactions import in_test_assert_no_transaction
+
+        in_test_assert_no_transaction(
+            f"remote service method to {self.path} called inside transaction!  Move service calls to outside of transactions."
+        )
+
+        if self.region:
+            target_mode = SiloMode.REGION
+        else:
+            target_mode = SiloMode.CONTROL
+
+        with SiloMode.exit_single_process_silo_context(), SiloMode.enter_single_process_silo_context(
+            target_mode, self.region
+        ):
+            content_type = f"application/json; charset={_RPC_CONTENT_CHARSET}"
```

Comment:
You could read the the content type from the provided headers
```suggestion
            content_type = headers.get('Content-Type')
```

🤷